presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -115,6 +116,7 @@
     public static final String FILE_RENAMING_ENABLED = "file_renaming_enabled";
     public static final String PREFER_MANIFESTS_TO_LIST_FILES = "prefer_manifests_to_list_files";
     public static final String MANIFEST_VERIFICATION_ENABLED = "manifest_verification_enabled";
+    public static final String METASTORE_USER_SUPPLIED_EXTRA_PARAM = "metastore_user_supplied_extra_param";
```

Comment:
Also if you change it to `Map` you can return an empty map by default so you don't have a "null" problem =D